types/vuelidate/vuelidate.d.ts

```diff
@@ -8,20 +8,30 @@ import Vue, { PluginFunction } from 'vue'
  */
 export const validationMixin: any
 
-// const Validation
-export interface Validation extends Vue {
-    // const validationGetters
-    readonly $invalid: boolean
-    readonly $dirty: boolean
-    readonly $error: boolean
-    readonly $pending: boolean
-    readonly $params: { [attr: string]: any }
-    $model: any
+export interface Validation {
+  readonly $anyDirty: boolean;
+  readonly $anyError: boolean;
+  readonly $dirty: boolean;
+  readonly $error: boolean;
+  readonly $invalid: boolean;
+  readonly $params: { [attr: string]: Params | null };
+  readonly $pending: boolean;
+  $model: any;
+  $touch(): never;
+  $reset(): never;
+  $flattenParams(): ValidationParams[];
+  // then we need an extension where any ADDITIONAL property to those specified above returns
+  // a Validation | boolean - I am not sure how to achieve this, so instead
+  // we have to include all the other types returned above
+  [attr: string]: any; // Validation | boolean;
+}
+
+interface EachValidation extends Validation {
+  $iter: Validation;
+}
 
-    // const validationMethods
-    $touch(): never
-    $reset(): never
-    $flattenParams(): ValidationParams[]
+export interface IterableValidation extends Validation {
+  $each: EachValidation;
```

Comment:
Either all Validation have $iter or null or you come here be unchecked casting. What speaks against putting each on all Validation?

Reply:
It is better to put it as possibly undefined on a single interface as you suggest - change added for your perusal.